src/it/resources-noncompilable/org/checkstyle/suppressionxpathfilter/missingnullcaseinswitch/InputXpathMissingNullCaseInSwitchNested.java

```diff
@@ -0,0 +1,22 @@
+// non-compiled with javac: Compilable with Java21
+package org.checkstyle.suppressionxpathfilter.missingnullcaseinswitch;
+
+public class InputXpathMissingNullCaseInSwitchNested {
+    void test(Object obj) {
+       int x =  switch (obj) {
+            case Rectangle(ColoredPoint _, ColoredPoint _) -> 1;
+            case ColoredPoint(int _, int _, String _) -> 2;
+            case null -> 3;
+            default -> {
+                int y = switch (obj) { // warn
+                    case Rectangle(ColoredPoint _, ColoredPoint _) -> 1;
+                    case ColoredPoint(int _, int _, String _) -> 2;
+                    default -> 4;
```

Comment:
Doesn't the default case cover the null case too? This code has no issues of a NPE because of the default right?

Reply:
I tested this on my local, and default does not cover the null case. I find this a bit odd TBH.

Please add this to the documentation.

Reply:
It seems if you don't add a `null` case, Java adds a `Objects.requireNonNull` before the switch. With a `null` case, that code isn't inserted. In my test bed, a `null` case was represented as `-1`.

Reply:
> Please add this to the documentation.

can you clarify what to add exactly? I think It is clear that `case null` is required to eliminate the risk of NPE, not the default.  as in this example  in the doc
```Java
 void testPatterns(Object obj) {
    // violation below, 'Switch using reference types should have a null case.'
    switch (obj) {
      case Integer i : {} break;
      default : {}
    }
  }
```

Reply:
Maybe it is just me, but it was not clear that `default` doesn't cover the `case null`. I would assume most people think `default` covers all cases not explicitly defined. Now it is like it "covers all cases not explicitly defined excluding `null` cases".

Reply:
It is different to keep compatibility.
https://stackoverflow.com/a/72475165/1015848